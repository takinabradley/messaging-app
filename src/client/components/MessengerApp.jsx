import { Link } from "react-router-dom"
import redirectIfLoggedOut from "../modules/react-router/loaders/redirectIfLoggedOut"

export async function loader({ request, params }) {
  return await redirectIfLoggedOut({ request, params })
}

export default function MessengerApp() {
  return (
    <div className="MessengerApp" data-testid="MessengerApp">
      Welcome to the messenger app!
      <div>
        wanna go to the <Link to="/">main page</Link>
      </div>
      <div>
        wanna <Link to="/logout">log out</Link>?
      </div>
      <div>
        view <Link to={"/account"}>account settings</Link>
      </div>
    </div>
  )
}
