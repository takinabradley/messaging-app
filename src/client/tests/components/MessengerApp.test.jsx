import { describe } from "vitest";
import MessengerApp from "../../components/MessengerApp";
import { render, screen } from "@testing-library/react";
import { RouterProvider, createMemoryRouter } from "react-router-dom";

describe("MessengerApp", () => {
    test("renders", async () => {
        const router = createMemoryRouter([
            {
                path: "/",
                element: <MessengerApp />
            }
        ])

        render(<RouterProvider router={router} />)

        expect(screen.getByTestId("MessengerApp")).toBeInTheDocument()
        expect(screen.getByTestId("MessengerApp")).toMatchSnapshot()
    })
})