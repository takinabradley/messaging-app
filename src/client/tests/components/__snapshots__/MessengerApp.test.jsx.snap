// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MessengerApp > renders 1`] = `
<div
  class="MessengerApp"
  data-testid="MessengerApp"
>
  Welcome to the messenger app!
  <div>
    wanna go to the 
    <a
      href="/"
    >
      main page
    </a>
  </div>
  <div>
    wanna 
    <a
      href="/logout"
    >
      log out
    </a>
    ?
  </div>
  <div>
    view 
    <a
      href="/account"
    >
      account settings
    </a>
  </div>
</div>
`;
